### PR TITLE
Finish feature selection functionality

### PR DESCRIPTION
**Pull request by stdavis**
highlight selected feature in grid.
select row on map click
fix bug related to selecting features of different geometry types
hide features tab by default
populate feature details on selection

### CODE REVIEW

File: src/app/project/tests/FeatureDetailsTests.html [hunk 1]
```diff
@@ -24,7 +24,7 @@
 
 require({
     baseUrl: baseUrl,
-    packages: ['app', 'bootstrap-stylus', 'esri', 'dojo', 'dijit', 'dojox', 'xstyle',
+    packages: ['app', 'bootstrap-stylus', 'esri', 'dojo', 'dijit', 'dojox', 'xstyle', 'mustache',
```
Comment (steveoh) [inline src/app/project/tests/FeatureDetailsTests.html]: this should be a direct dependency in bower.

**Comment by steveoh** [comment]
featuredetails has a memory leak on it's topic subscription which you can notice if you delete and recreate the widget by selecting more than one project.

**Comment by steveoh** [comment]
can you also close the issue in the commit message.

**Comment by stdavis** [comment]
Got the three items done.

**Comment by steveoh** [comment]
closed with d821c7a

**Pull request closed without merging**